File: cpp/04-gtest-mock/src/Money.cpp
#include "Money.hpp"

Money::Money(double amount, const std::string& currency) : amount(amount), currency(currency) {}

double Money::getAmount() const {
    return amount;
}

std::string Money::getCurrency() const {
    return currency;
}

std::string Money::toString() const {
    std::ostringstream oss;
    oss << std::fixed << std::setprecision(2) << amount << " " << currency;
    return oss.str();
}

File: java/04-with-mock/src/main/java/com/foo/currency/Money.java
package com.foo.currency;

import java.math.BigDecimal;
import java.util.Currency;

/**
 * 货币金额类
 */
public class Money {
    private final BigDecimal amount;
    private final Currency currency;
    
    public Money(BigDecimal amount, Currency currency) {
        this.amount = amount;
        this.currency = currency;
    }
    
    public BigDecimal getAmount() {
        return amount;
    }
    
    public Currency getCurrency() {
        return currency;
    }
    
    @Override
    public String toString() {
        return String.format("%.2f %s", amount, currency.getCurrencyCode());
    }
}
